src/plugins/workspace/server/workspace_client.ts

```diff
@@ -4,19 +4,50 @@
  */
 
 import { i18n } from '@osd/i18n';
-import type {
+import {
   SavedObject,
   SavedObjectsClientContract,
   CoreSetup,
   WorkspaceAttribute,
   SavedObjectsServiceStart,
+  ACL,
 } from '../../../core/server';
 import { WORKSPACE_TYPE } from '../../../core/server';
-import { IWorkspaceClientImpl, WorkspaceFindOptions, IResponse, IRequestDetail } from './types';
+import {
+  IWorkspaceClientImpl,
+  WorkspaceFindOptions,
+  IResponse,
+  IRequestDetail,
+  WorkspaceAttributeWithPermission,
+  WorkspacePermissionItem,
+} from './types';
 import { workspace } from './saved_objects';
 import { generateRandomId } from './utils';
 import { WORKSPACE_SAVED_OBJECTS_CLIENT_WRAPPER_ID } from '../common/constants';
 
+const convertToACL = (
```

Comment:
Any reason that we do the transformer in server side? What about doing that in browser side so that the type in server/router will comply with the type of `Permissions`?